### PR TITLE
Enabling conda packaging in CI builds

### PR DESCRIPTION
**Pull request by rmshaffer**
This PR enables conda packaging as part of the default CI build. It moves the creation of the self-contained executable into a separate build job called `pack_selfcontained`. It also removes `/property:PublishSingleFile=true` from the `dotnet publish` command such that each built assembly remains in a separate file to enable signing in official builds.

For an example of the CI results, see [this build](https://dev.azure.com/ms-quantum-public/Microsoft%20Quantum%20(public)/_build/results?buildId=11018&view=results).

### CODE REVIEW

File: build/steps-selfcontained.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+- task: UseDotNet@2
+  displayName: 'Use .NET Core SDK 3.1.100'
+  inputs:
+    packageType: sdk
+    version: '3.1.100'
+
+##
+# Pack
+##
+- pwsh: .\pack-selfcontained.ps1
```
Comment (cgranade) [inline build/steps-selfcontained.yml:33]: This shouldn't matter on `pwsh`, but suggesting for consistency with other changes in this PR.
```suggestion
- pwsh: ./pack-selfcontained.ps1
```

File: build/steps-selfcontained.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    downloadPath: '$(System.DefaultWorkingDirectory)/artifact'
+
+- pwsh: |
+    Move-Item "$Env:SYSTEM_DEFAULTWORKINGDIRECTORY/artifact/iqsharp/drops" "$Env:SYSTEM_DEFAULTWORKINGDIRECTORY/drops"
+  displayName: "Copy build artifacts to IQ# working directory"
+
+- task: CondaEnvironment@1
+  inputs:
+    packageSpecs: "python=3.7 pip setuptools pytest jupyter"
+  displayName: 'Use conda environment w/ Python 3.'
```
Comment (cgranade) [inline build/steps-selfcontained.yml:22]: Is this used by the self-contained job?

Comment (rmshaffer) [reply build/steps-selfcontained.yml:22]: I don't think so. I'll remove it and verify. Thanks for noticing this, will save a little bit of time in the build!

**Comment by rmshaffer** [comment]
> Re-approving for newest changes.

Thanks for the review, @cgranade! I'll wait to merge until I've verified another end-to-end QDK build with these latest changes.

**Comment by cgranade** [comment]
> Thanks for the review, @cgranade! I'll wait to merge until I've verified another end-to-end QDK build with these latest changes.

Makes perfect sense, I just wanted to make sure you were unblocked if the QDK build worked as expected. Thanks!

**Pull request merged**